robot/Cumulus/resources/GiftEntryPageObject.py

```diff
@@ -1,21 +1,119 @@
         self.selenium.go_to(url)
         self.salesforce.wait_until_loading_is_complete()
-        self.selenium.wait_until_page_contains("Templates")
\ No newline at end of file
+        self.selenium.wait_until_page_contains("Templates")
+
+    def _is_current_page(self):
+        """
+        Verifies that current page is Gift Entry landing page
+        """
+        self.selenium.wait_until_location_contains("GE_Gift_Entry", timeout=60, 
+                                                   message="Current page is not Gift Entry landing page")
+        self.selenium.wait_until_page_contains("Default Gift Entry Template")                                               
+
+    def click_gift_entry_button(self,title):
+        """clicks on Gift Entry button identified with title"""
+        locator=npsp_lex_locators["gift_entry"]["button"].format(title)
+        self.selenium.wait_until_page_contains_element(locator)
+        self.selenium.click_element(locator)  
+
+    def enter_value_in_field(self,**kwargs):
+        """Enter value in specified field"""
+        for key,value in kwargs.items():
+            if key=='Description':
+                locator=npsp_lex_locators["gift_entry"]["textarea"].format(key)
+                self.selenium.wait_until_page_contains_element(locator)
+                self.salesforce._populate_field(locator, value)
+            else:
+                locator=npsp_lex_locators["gift_entry"]["field_input"].format(key)
+                self.selenium.wait_until_page_contains_element(locator)
+                self.salesforce._populate_field(locator, value)      
+
+    def select_template_action(self,name,action):
+        """From the template table, select template with name and select an action from the dropdown"""
+        locator=npsp_lex_locators["gift_entry"]["actions_dropdown"].format(name)
+        self.selenium.click_element(locator)
+        element=self.selenium.get_webelement(locator)
+        status=element.get_attribute("aria-expanded")
+        if status=="false":
+            self.selenium.wait_until_page_contains("Clone")    
+        self.selenium.click_link(action)
+        if action=="Edit" or action=="Clone":
+            self.selenium.wait_until_page_contains("Gift Entry Template Information")
+        elif action=="Delete":
+            self.selenium.wait_until_page_does_not_contain(name)    
+
+    def select_object_group_field(self,object_group,field):
+        """Select the specified field under specified object group 
+           to add the field to gift entry form and verify field is added"""
+        locator=npsp_lex_locators["gift_entry"]["form_object_dropdown"].format(object_group)
+        self.selenium.scroll_element_into_view(locator)
+        self.selenium.click_element(locator)
+        element=self.selenium.get_webelement(locator)
+        status=element.get_attribute("aria-expanded")
+        if status=="false":
```

Comment:
Can you please explain this condition and sleep here

Reply:
when we click on the dropdown for object group, aria-expanded attribute will be set to true if dropdown expands, otherwise it will be false. so after clicking on dropdown, I am checking if its true or false. If true it will go ahead and perform next operation, incase if browser is slow in opening, it will be false so it will wait for 2 seconds for it to open and then try to perform next actions